oleksandrivantsiv/sonic-buildimage
# [202205] Clear /etc/resolv.conf after changing the mgmt interface configuration to static.

#### Why I did it
fixes #xxxx 

When the management interface IP address configuration changes from dynamic to static the DNS configuration (retrieved from the DHCP server) in /etc/resolv.conf remains uncleared. This leads to a DNS configuration pointing to the wrong nameserver.  

#### How I did it

Use resolvconf package for managing DNS configuration. It is capable of tracking the source of DNS configuration and puts the configuration retrieved from the DHCP servers into a separate file. This allows implementation cleanup of DNS configuration retrieved from DHCP.

#### How to verify it

1. Ensure that the management interface has no static configuration.
2. Check that /etc/resolv.conf has DNS configuration.
3. Configure a static IP address on the management interface.
4. Verify that /etc/resolv.conf has no DNS configuration.
5. Remove the static IP address from the management interface.
6. Verify that /etc/resolv.conf has DNS configuration retrieved form DHCP server.

#### Which release branch to backport (provide reason below if selected)




#### Description for the changelog


#### Ensure to add label/tag for the feature raised. example - PR#2174 under sonic-utilities repo. where, Generic Config and Update feature has been labelled as GCU.

#### Link to config_db schema for YANG module changes


#### A picture of a cute animal (not mandatory but encouraged)
